src/match_string.h

```diff
@@ -15,6 +13,38 @@
  * index of a @string in the @array if matches, or %-EINVAL otherwise.
  */
 
+/*@
+    logic integer cmp(unsigned char a, unsigned char b) =
+       a == b ? 0 : a < b ? -1 : 1;
+
+    logic integer strncmp(char *cs, char *ct, integer n) =
+       n == -1 ? 0 : (cs[n] == ct[n] ? strncmp(cs+1, ct+1, n-1) : cmp((u8 AENO)cs[n], (u8 AENO)ct[n]));
+    logic integer strcmp(char *cs, char *ct) = strncmp(cs, ct, strlen(cs));
+*/
+
+/*@
+    requires \valid_read(array + (0..n-1));
+    requires valid_str(string);
+    requires n <= INT_MAX;
+    requires \forall integer i; 0 <= i < n && valid_str(array[i]);
```

Comment:
These requirements (28-29) are too strict.
According to the documentation n could be -1 ("@n:	number of strings in the array or -1 for NULL terminated arrays"). You need to reflect this in line 28. It's possible to have null in array, it should be reflected in 29.